### DISCUSSION TITLE
Author attribution for github repo

### DISCUSSION THREAD

**Discussion by jerhill** [community/community#31410] [Other Feature Feedback, Questions, & Ideas]
I have recently published a repo that I am the author for >90% of the code, but github puts two authors ahead of me when I search for it, and this has percolated to the doi. I don't in general make a big deal about authorship priority , but I'd like to be able to point to something that I made and not be the third author on the reference.
How does github assign authorship, and how can I fix this?

-j

**Comment by airtower-luna** [comment]
> but github puts two authors ahead of me when I search for it

What do you mean by "assigning authorship", where does that happen?

In general the relevant attribution for source code is what's in the source itself, e.g. a copyright and license info header in a source file, or a contributors list stored in a dedicated file. If someone's ordering authors on a paper by some random listing on a repository page or similar instead I'd say they're doing a really bad job.

If you mean recognizing commits as yours: The answer is simply that it happens by the author email recorded in the commit. The email address must be on your account to be recognized as yours.
